build.sbt

```diff
@@ -202,3 +204,13 @@ lazy val dumpPersistenceSchemaCompilerPlugin = (projectMatrix in file("dump-pers
     assembly / assemblyJarName := s"${name.value}_${scalaBinaryVersion.value}-${version.value}.jar", //Warning: this is a default name for packageBin artefact. Without explicit rename of packageBin will result in race condition
     addArtifact(Compile / assembly / artifact, assembly))
   .jvmPlatform(scalaVersions = supportedScalaVersions)
+
+lazy val checkCurrentJavaVersion =
+  taskKey[Unit]("checks whether SBT has been started with proper java version (at least 11)")
+
+checkCurrentJavaVersion := {
+  if (sys.props.getOrElse("java.specification.version", "0").toDouble < 11.0) {
```

Comment:
Does the project compile with no issues on Java 14/17?

Reply:
Yes, it compiles with no issues with both 14 and 17 - so in my opinion it is ok to say that JDK version should be at least 11 (might be higher)